docs/standards/v1/STANDARDS.md

```diff
@@ -0,0 +1,322 @@
+
+## Response Codes
+
+### Successful Requests
+
+|Status Code|Description|Verbs|
+|---|---|---|
+|200 OK|This status **MUST** be returned for synchronous requests that complete successfully and have a response body. This **MUST** only be used if there is not a more appropriate 2XX response code. |GET, PATCH|
+
+### Client Errors
+
+|Status Code|Description|Verbs|
+|---|---|---|
+|400 Bad Request|This status **MUST** be returned for requests that provide malformed or invalid data. Examples: malformed request body, unexpected query parameters, or invalid request fields.|GET, PATCH, POST, DELETE|
+|403 Forbidden|This status **MUST** be returned if the request cannot be performed by the user due to lack of permissions. Example: User with read-only permissions to a resource tries to update it. |POST, PATCH, DELETE|
+|404 Not Found|This status **MUST** be returned if the requested resource does not exist or if the user requesting the resource has insufficient permissions to view the resource.|GET, POST, PATCH, DELETE|
+
+### Server Errors
+
+|Status Code|Description|
+|---|---|
+|500 Internal Server Error|This status **MUST** be returned when a server error occurs.
+
+## Resources
+
+A resource represents an individual object within the system.  It is represented as a JSON object.  
+
+## Field Names
+
+Resource Fields will be camelCase.
+
+Resource Fields MUST include ONLY the following characters:
+
+* `a-z` (lowercase alpha)
+* `A-Z` (uppercase alpha)
+
+Resource fields that accept multiple values MUST be pluralized.
+
+#### Recommended Technologies to Support
+
+* https://github.com/vbabiy/djangorestframework-camel-case
+
+## Collections
+
+A collection is a list of multiple Resources. A collection is represented as a JSON object.
+
+If a collection is larger then per_page Pagination size then it `MUST` contain a links object, containing pagination information.
+
+
+## Pagination
+
+Pagination should be used by Collections to limit the number of resources returned at a time.
+
+Pagination is requested by a client through the use of query parameters.
+
+Requests will accept the following query parameters:
+
+- limit: maximum number of items to return
+- offset: starting position of the query in relation to the complete set of unpaginated items
+
+A paginated response would look as follows:
+
+```json
+{
+    "count": 1023,
+    "next": "https://api.example.org/accounts/?limit=100&offset=500",
+    "previous": "https://api.example.org/accounts/?limit=100&offset=300",
+    "results": [
+       …
+    ]
+}
+```
+#### Recommended Technologies to Support
+
+To achieve this, register a `DEFAULT_PAGINATION_CLASS` in your `local_settings/settings.py` to apply to all apps:
+
+```python
+REST_FRAMEWORK = {
+    "DEFAULT_PAGINATION_CLASS": "rest_framework.pagination.LimitOffsetPagination",
+    "PAGE_SIZE": 10
+}
+```
+
+PAGE_SIZE should be set to:  "TO BE DETERMINED"
```

Comment:
I think it's going to be hard to come up with a one-size-fits-all default given that it will vary on resource size and query complexity, but I certainly think something larger than 10 should be the recommendation 😄

Reply:
https://github.com/hackoregon/2019-backend-cookiecutter-django/commit/37aa17e08fb80bad6deee2fe21e39db2a4419e55